src/python/pants/jvm/goals/coursier.py

```diff
@@ -110,21 +107,13 @@ class CoursierGenerateLockfileResult:
 async def coursier_generate_lockfile(
     request: CoursierGenerateLockfileRequest,
     jvm: JvmSubsystem,
-    resolves_to_addresses: JvmResolvesToAddresses,
+    resolves_to_artifacts: JvmResolvesToArtifacts,
 ) -> CoursierGenerateLockfileResult:
-    # Find JVM artifacts in the dependency tree of the targets that depend on this lockfile.
-    # These artifacts constitute the requirements that will be resolved for this lockfile.
-    dependee_targets = await Get(
-        TransitiveTargets, TransitiveTargetsRequest(resolves_to_addresses.get(request.resolve, ()))
-    )
-    artifact_requirements = ArtifactRequirements(
-        ArtifactRequirement.from_jvm_artifact_target(tgt)
-        for tgt in dependee_targets.closure
-        if JvmArtifactFieldSet.is_applicable(tgt)
-    )
-
     resolved_lockfile = await Get(
-        CoursierResolvedLockfile, ArtifactRequirements, artifact_requirements
+        CoursierResolvedLockfile,
+        ArtifactRequirements,
+        # TODO: error if no artifacts associated with a resolve?
```

Comment:
I think we probably should allow it? You'd have an empty lockfile generated, which is a little weird but also legal.

Reply:
At the very least, it is common in tests.

Reply:
Yeah we should allow it. We have in pantsbuild/pants a `jvm_testprojects` resolve. Currently, it's unused, but it's convenient we have it defined already in `pants.toml` so that we can add to it.

Reply:
> Yeah we should allow it. We have in pantsbuild/pants a `jvm_testprojects` resolve. Currently, it's unused, but it's convenient we have it defined already in `pants.toml` so that we can add to it.

It's the default resolve, so it's not unused... it's used for the code under `testprojects/src/jvm`, right?

Reply:
They don't use any `jvm_artifact`s. Note that the lockfile has always been empty, and remains empty with this change

Reply:
Ah, sure. Not unused, just empty.